### PR TITLE
Don't ignore player joins/leaves

### PR DESCRIPTION
**Pull request by AnnikaCodes**
Implements [this suggestion](https://www.smogon.com/forums/threads/turning-on-ignore-spectators-shouldnt-prevent-me-from-seeing-my-opponents-leave-join-messages.3667838/).

### CODE REVIEW

File: src/battle.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ type WeatherState = [string, number, number];
 type EffectTable = {[effectid: string]: EffectState};
 type HPColor = 'r' | 'y' | 'g';
 
+const PLAYER_GROUP = '\u2606';
```
Comment (Zarel) [inline src/battle.ts:37]: How does the ignoreSpects code work usually? Presumably you only want to ignore unauthed users and maybe voices.

**Comment by Zarel** [comment]
It looks like it's in `battle-log`. Can you implement this inside `battle-log` as well?

File: src/battle.ts (hunk 1 above)
Comment (AnnikaCodes) [reply src/battle.ts:37]: Yes, exactly -- but the logic for displaying `|j|` and `|l|` messages didnt respect that.

**Comment by Zarel** [comment]
Your `battle.ts` changes are no longer necessary, right?

**Comment by AnnikaCodes** [comment]
> Your `battle.ts` changes are no longer necessary, right?

I'm not sure -- I'll test it out locally.

**Comment by AnnikaCodes** [comment]
turns out I messed smth up, this is working now

**Comment by Zarel** [comment]
Great, thanks!

**Pull request merged**